proposals/053-cryptography-extension.md

```diff
@@ -0,0 +1,124 @@
+
+These issues and confounding factors could have been handled by better prioritization and scope management, and a more clear communication of any effect on the project timeline. 
+
+However, there are mitigating factors as well: significant progress was made towards a high-value stretch goal (Extended X509 suppport), and the deliverables were met despite the size and scope of the project being significantly larger than originally estimated.
+
+As such, this proposal is slightly less aggressive in scope, to allow for a little more flexibility to account for shifting needs and feedback.
+
+# Technical Content
+
+There is still more work to do, as the scope of the original proposal was deliberately restricted for the purposes of focus and achievability. Now that the original goals have been met, some of the previously out-of-scope goals may now be reconsidered.
+
+## Continued improvements to `botan`
+
+While `botan-bindings` and `botan-low` are effectively frozen, small changes may still occur. However, `botan` is likely to see significant churn as we continue to improve and respond to feedback.
+
+As `botan` is a very "fresh" library, it is expected to see some amount of churn in its API as it is developed and refined further. The initial goals of functionality are nominally complete, but there is still a great amount of ways to improve the safety and user experience.
+
+Expected improvements include:
+
+- Managed / limited lifespan objects with bracket / guaranteed cleanup
+- Auto-zeroing of sensitive memory
+- Implementing data types and instances for libraries such as `random` and `mtl`
+- Conditional compilation suport for optional modules
+- Improvements to the `Make / Remake` factory methods used to generate bindings
+
+## Botan C++ development
+
+`Botan C++` is being actively developed, so this project will need continued maintenance in order to achieve continued parity. Furthermore, several valuable features in `Botan C++` are not exposed by the C FFI, and require C / C++ development in order to expose them:
+
+- Stream ciphers
+- Complete X509 infrastructure implementation
+- Complete TLS server implementation
+
+One of the previous stretch goals was to develop the C bindings for X509 certificate architecture, and to [contribute that code back](https://github.com/randombit/botan/issues/3627) to `Botan C++`. Substantial progress was made, but further work is required in order to complete this. 
+
+Another feature missing from the Botan C FFI is Stream Ciphers, and like the X509 bindings, it will require effort to extend the Botan C++ library in order to expose it to Haskell.
+
+A third feature missing from the Botan C FFI is TLS, as Botan provides a full TLS server implementation, which will also require effort to expose to the FFI.
```

Comment:
I understand the rationale, but essentially the proposal asks HF to sponsor a development of a third-party C library, which we have no control of / do not gain any stake in / are at mercy of its leadership. This is a high risk investment, extremely difficult to manage. Could we rather focus on lower-hanging fruits and wait (potentially indefinitely) until someone else does Botan C FFI?